app/shared/controllers/validators/resolvers/ResolveDateRange.scala

```diff
@@ -16,60 +16,56 @@
 
 package shared.controllers.validators.resolvers
 
-import shared.models.domain.DateRange
-import shared.models.errors.{EndDateFormatError, MtdError, RuleEndBeforeStartDateError, StartDateFormatError}
 import cats.data.Validated
 import cats.data.Validated.{Invalid, Valid}
 import cats.implicits._
+import shared.models.domain.DateRange
+import shared.models.errors.{EndDateFormatError, MtdError, RuleEndBeforeStartDateError, StartDateFormatError}
 
 import java.time.LocalDate
+import scala.math.Ordering.Implicits.infixOrderingOps
 
-private[resolvers] class ResolveDateRange private (yearLimits: Option[YearLimits]) extends Resolver[(String, String), DateRange] {
+case class ResolveDateRange(startDateFormatError: MtdError = StartDateFormatError,
+                            endDateFormatError: MtdError = EndDateFormatError,
+                            endBeforeStartDateError: MtdError = RuleEndBeforeStartDateError)
+    extends ResolverSupport {
 
-  def apply(value: (String, String), notUsedError: Option[MtdError], path: Option[String]): Validated[Seq[MtdError], DateRange] = {
-    val (startDate, endDate) = value
+  private def resolveDateRange(parsedStartDate: LocalDate, parsedEndDate: LocalDate): Validated[Seq[MtdError], DateRange] =
+    if (parsedEndDate <= parsedStartDate)
+      Invalid(List(endBeforeStartDateError))
+    else
+      Valid(DateRange(parsedStartDate, parsedEndDate))
 
-    val resolvedDates = (
-      ResolveIsoDate(startDate, StartDateFormatError),
-      ResolveIsoDate(endDate, EndDateFormatError)
+  val resolver: Resolver[(String, String), DateRange] = { case (startDate, endDate) =>
+    (
+      ResolveIsoDate(startDate, startDateFormatError),
+      ResolveIsoDate(endDate, endDateFormatError)
     ).mapN(resolveDateRange).andThen(identity)
-
-    yearLimits match {
-      case Some(YearLimits(minYear, maxYear)) => resolvedDates.andThen(validateFromAndToDate(_, minYear, maxYear))
-      case None                               => resolvedDates
-    }
   }
 
-  private def resolveDateRange(parsedStartDate: LocalDate, parsedEndDate: LocalDate): Validated[Seq[MtdError], DateRange] = {
-    val startDateEpochTime = parsedStartDate.toEpochDay
-    val endDateEpochTime   = parsedEndDate.toEpochDay
+  def apply(value: (String, String)): Validated[Seq[MtdError], DateRange] = resolver(value)
 
-    if ((endDateEpochTime - startDateEpochTime) <= 0) {
-      Invalid(List(RuleEndBeforeStartDateError))
-    } else {
-      Valid(DateRange(parsedStartDate, parsedEndDate))
-    }
-  }
+  def dateLimited(minDate: LocalDate, maxDate: LocalDate): Resolver[(String, String), DateRange] =
+    resolver thenValidate ResolveDateRange.dateLimits(minDate, startDateFormatError, maxDate, endDateFormatError)
```

Comment:
Yes. It's not quite right. 

A couple of more obvious possibilities:

- Change the name of `thenValidate` to something else.  The result of the `resolver thenValidate ..` is another resolver, so could use `resolver withValidation ...` maybe? It doesn't convey the order that validations happen as directly but it might read better. 

- Change the name of the `resolver` val e.g. to `resolve` so you'd have `resolve thenValidate ...`. Trouble is the type is a `Resolver[...]` (not a `Resolve[...]`) but maybe that's not a problem. The classes/objects that would expose these `resolve` members would then start appearing in code as `ResolveDate.resolve thenValidate ...`, which isn't too bad. Would probably want to rename `resolverOptionally` and `resolveOptionallyWithDefault` to simply `optionally` and `optionallyWithDefault` respectively.

WDYT?